rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -39,60 +41,77 @@
     "roberta": 512,
 }
 
-logger = logging.getLogger(__name__)
-
 
-class LanguageModelFeaturizer(DenseFeaturizer):
-    """Featurizer using transformer-based language models.
+class LanguageModelFeaturizerGraphComponent(DenseFeaturizer2, GraphComponent):
+    """A featurizer that uses transformer-based language models.
 
-    The transformers(https://github.com/huggingface/transformers) library
-    is used to load pre-trained language models like BERT, GPT-2, etc.
-    The component also tokenizes and featurizes dense featurizable attributes of
+    This component loads a pre-trained language model
+    from the Transformers library (https://github.com/huggingface/transformers)
+    including BERT, GPT, GPT-2, xlnet, distilbert, and roberta.
+    It also tokenizes and featurizes the featurizable dense attributes of
     each message.
     """
 
-    defaults = {
-        # name of the language model to load.
-        "model_name": "bert",
-        # Pre-Trained weights to be loaded(string)
-        "model_weights": None,
-        # an optional path to a specific directory to download
-        # and cache the pre-trained model weights.
-        "cache_dir": None,
-    }
+    def __init__(
+        self, config: Dict[Text, Any], execution_context: ExecutionContext,
+    ) -> None:
+        """Initializes the featurizer with the model in the config."""
+        super(LanguageModelFeaturizerGraphComponent, self).__init__(
+            execution_context.node_name, config
+        )
+        self._load_model_metadata()
+        self._load_model_instance()
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns LanguageModelFeaturizer's default config."""
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            # name of the language model to load.
+            "model_name": "bert",
+            # Pre-Trained weights to be loaded(string)
+            "model_weights": None,
+            # an optional path to a specific directory to download
+            # and cache the pre-trained model weights.
+            "cache_dir": None,
+            "language": None,
+            # TODO: the supported language should depend on the language the model
+            #  was pre-trained on. In the future add this check to config validation.
+        }
 
     @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        """Packages needed to be installed."""
-        return [Tokenizer]
+    def validate_config(cls, config: Dict[Text, Any]) -> None:
```

Comment:
@ka-bu What do you think of doing this with Python Protocols in the future? Too many of the components implement this with `pass`. I think using protocols instead would make this more flexible

Reply:
I'm not sure. I like the idea of protocols, but the nice thing about the abstract method is that it forces everyone to think about this and implement it. With the protocol we might miss some essential validation without even noticing it (although that's not really a problem with `validate_config` - was just thinking of the other `validate_..` where we replaced tokenizer by tokenizer type or so). 
And if we use the `validate_config` in all components eventually then it doesn't make a difference really whether it's a protocol or there's a version in `GraphComponent` that just passes, no?

Reply:
Good points 👍🏻 Let's just see how this develops and find the right abstractions then 👍🏻